src/sinks/util/buffer/metrics.rs

```diff
@@ -17,8 +19,9 @@ impl Eq for MetricEntry {}
 impl Hash for MetricEntry {
     fn hash<H: Hasher>(&self, state: &mut H) {
         let metric = &self.0;
-        std::mem::discriminant(&metric.value).hash(state);
+        discriminant(&metric.value).hash(state);
         metric.name.hash(state);
+        metric.namespace.hash(state);
```

Comment:
Just noticing this now, maybe we could add a test to ensure the namespace is taken into consideration for hashing?

Reply:
I'll see what I can do.

Reply:
I'm not sure I can do so in a way that is guaranteed to continue working if the default hash function in Rust changes its implementation or seed. As it is a lossy compression function, the hash value is allowed to be equal for distinct values, which is why the `eq` function needed rewriting in the first place.

Reply:
Could we have a test that just asserts the hash is different after only changing the namespace value?

Reply:
The hash will *probably* will be different, but it might not be always, making it only a mostly reliable test.

Reply:
Hmm, true. I _think_ we could control for that by passing in a dummy `Hasher` that is simply an identity function?

Not a huge deal, I'm just wondering about preventing regressions here.